### ISSUE TITLE
[masthead]: IBM logo not visible in Windows high contrast mode

### ISSUE THREAD

**Issue by kennylam** [carbon-design-system/carbon-for-ibm-dotcom#10977]
### Description

The IBM logo in the `masthead` component is not visible in Windows high contrast mode.

### Component(s) impacted

`masthead`

<img width="743" alt="image" src="https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/909118/525a2d71-0608-47e1-8f23-9ac7b393df39">

### Browser

Firefox, Microsoft Edge

### Carbon for IBM.com version

1.51.0

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components, @carbon/ibmdotcom-styles

### CodeSandbox example

https://www.ibm.com

### Steps to reproduce the issue (if applicable)

In Windows, enable high contrast viewing mode. Visit https://www.ibm.com

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
